### DISCUSSION TITLE
StLink cannot connect to target.

### DISCUSSION THREAD

**Discussion by Maestroo420** [timeylies/How-to-convert-a-Ninebot-SNSC-2.2-or-2.3-Scooter#1] [Issues]
Hello, i've been sitting here for hours trying to connect my ecu to stlink via a stlink. It keeps popping up " can't connect to target " I've changed everything everyone told me to change, still no luck. Some people say I have a fake(clone?) stlink. It's from Amazon so I wouldn't be surprised, they also say stlink will recognize if it is indeed a fake one. If that's my issue is there any other software that supports non genuine stlinks? BIIIIIIG Regards. M

**Comment by Mysterious-Wrangler** [comment]
I heard the program used here is kinda outdated by now, maybe thats why the issue is happening, i dont know. Anyway did you try to disable readout protection and then try to connect? The tutorial says to try to enable it but its enabled by default and you should instead try to disable it, thats what worked for me anyway. But im having some troubles myself after the flashing so i dont know anymore from that. Good luck.

**Comment by timeylies** [comment]
it’s maybe because of the st link you used, I do recommend trying with the st link I mentioned in the wiki

EDIT: Check your connections first. If it says "Can't connect to target," that usually means the st-link works, and the st-link can't communicate to the mcu.

**Reply by Maestroo420** [reply]
Okay okay.. i succesfully flashed the esc. Now in onto the dash, i've tried flashing the dash that comes with the scooter (VOI snsc 2.3) Doesnt work ( most likely Skill diff). Now im trying to flash BLE555 to a Xiaomi m365 dash bc it has the same Connectors etc. Is it even possible to flash ble555 on a m365 dash? It cant Connect to target.. sometimes it recognize it but disconnects as soon as it connects. Any tips? Ideas? REGARDS

**Reply by timeylies** [reply]
Yeah it should be possible to flash it, I don't have much ideas, if it doesn't work, buy another dash

**Reply by Maestroo420** [reply]
Alright! I've made some progress. Last question before i give up. I have connected my m365 dash ( with no luck flashing) I get the scooter turned on but taillight keeps blinking then the scooter turns off. Is that because the m365 isn't flashed with BLE555? Or is it another problem? Thanks

**Reply by timeylies** [reply]
If BLE555 doesn't work, you should try BLE115

**Reply by Maestroo420** [reply]
Okay, the problem im having whilst trying to flash the dash is that it won't connect. I've tried stlink utility, pgrCUBE, I even downloaded a ready to go openOCD script but it won't connect no matter how hard I'm trying. I'm so lost. The scooter is just blinkin, no response on brakes neither the throttle. Do I NEED the gps module hooked up? Or can I just let that cable be there. THANKS again, seriously

**Reply by timeylies** [reply]
Oh, then I have no ideas, usually that means connections but I'm not sure anymore

**Reply by Maestroo420** [reply]
Okay, do you have Any info about the GPS box? Do i need it pluggeg in?

**Reply by timeylies** [reply]
No you don't need it plugged in, besides the retail firmware isn't compatible with the iot

**Reply by Maestroo420** [reply]
Laaaaaaaaaaaast question, will this ST-link work?     ![image](https://github.com/user-attachments/assets/5d398f6a-8b35-4d85-b121-53b8226699af)

**Reply by timeylies** [reply]
Yeah It should. I suppose any stlink works, as long as it supports swd

**Reply by Maestroo420** [reply]
Thanks man
